test/withStyles_test.jsx

```diff
@@ -210,5 +210,37 @@ describe('withStyles()', () => {
       expect(Wrapped.propTypes).not.to.equal(MyComponent.propTypes);
       expect(Wrapped.defaultProps).not.to.equal(MyComponent.defaultProps);
     });
+
+    it('extends React.Component', () => {
+      function MyComponent() {
+        return null;
+      }
+
+      const Wrapped = withStyles(() => ({}))(MyComponent);
+      expect(Object.getPrototypeOf(Wrapped) === React.Component)
+        .to.equal(true);
+      expect(Object.getPrototypeOf(Wrapped.prototype) === React.Component.prototype)
```

Comment:
`expect(Object.getPrototypeOf(Wrapped.prototype)).to.equal(React.Component.prototype)` will have a nicer failure message, same throughout